src/nodeProvider.ts

```diff
@@ -89,21 +88,39 @@ export class DidactNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 		this._onDidChangeTreeData.fire(undefined);
 	}
 
-	public getParent(element : TreeNode) : TreeNode | undefined {
+	public async getParent(element : SimpleNode) : Promise<SimpleNode | null> {
+		console.log(`getting parent for ${element.toString()}`);
 		if (element instanceof TutorialNode) {
 			const tutorial = element as TutorialNode;
 			if (tutorial.category) {
-				return this.findCategoryNode(tutorial.category);
+				return Promise.resolve(this.findCategoryNode(tutorial.category));
 			}
 		}
-		return undefined;
+		if (element instanceof HeadingNode) {
+			const heading = element as HeadingNode;
```

Comment:
Sonar is reportign that the "This assertion is unnecessary since it does not change the type of the expression.sonarlint(typescript:S4325)"